test/specs/lib/reactComponentUtils-test.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react'
+import { getComponentName, areTypeNamesEqual } from 'src/lib/reactComponentUtils'
+import { createMockComponent } from 'test/utils'
+
+describe('getComponentName', () => {
+  it('returns null for invalid arguments', () => {
+    const invalidArgs = [undefined, null, '', {}]
+    invalidArgs.map(invalidArg => expect(getComponentName(invalidArg as any)).toBeFalsy())
+  })
+
+  it('returns the string component itself', () => {
+    const stringArgs = ['div', 'span', 'MyComponent']
+    stringArgs.map(stringArg => expect(getComponentName(stringArg)).toEqual(stringArg))
+  })
+
+  it('returns the function name', () => {
+    const MyComponent = createMockComponent('MyComponent')
+
+    function fooComponent() {
+      return null
+    }
+
+    expect(getComponentName(fooComponent)).toEqual('fooComponent')
+    expect(getComponentName(MyComponent)).toEqual('MyComponent')
+  })
+})
+
+describe('areTypeNamesEqual', () => {
+  it('returns false for invalid arguments', () => {
+    expect(areTypeNamesEqual(null, undefined)).toBeFalsy()
+    expect(areTypeNamesEqual(undefined, '')).toBeFalsy()
+    expect(areTypeNamesEqual('', '')).toBeFalsy()
+  })
+
+  describe('for primitive components', () => {
```

Comment:
rather `for primitive component types`, not just `components` (which implies instances)

Reply:
N.A. anymore since the implementation was drastically simplified following decision:
https://github.com/stardust-ui/react/pull/496#discussion_r238708722